### PR TITLE
feat: debugger api pane navigation

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description
Errors associated with a field i.e it points to a specific property path, clicking on it will navigate to the respective tab and scrolls to the field. If the errors does not have the property path it will simply navigate to the api page. This applies to logs and errors. Api execution errors simply navigates to the pane as they don't have a specific property path associated. 

Update data attribute name from replay to location id so that it is more generic. Updated the data attribute ids for pagination fields as they were incorrect and wasn't working for undo/redo as well. 

This PR works for api pane which includes rest api and graql api.  By default for other actions it will just navigate to the page. For Query errors/logs is handled in the next PR https://github.com/appsmithorg/appsmith/pull/25853. 

Skipped running tests for airgap, the tests will be updated and enabled for it separately. 

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/25464

#### Media

https://github.com/appsmithorg/appsmith/assets/67054171/c39cb220-0f86-4af2-b7c0-3f41b0695d15

#### Type of change
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
#### How Has This Been Tested?
- [x] Manual
- [ ] Jest
- [x] Cypress
>
>
#### Test Plan
https://github.com/appsmithorg/TestSmith/issues/2456
>
>
#### Issues raised during DP testing
https://github.com/appsmithorg/appsmith/pull/25771#issuecomment-1653448087
>
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [x] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [x] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by kavitasmoolya** [comment]
@akash-codemonk Please look into below issues 

- [ ] Header tab : Wrong key value pair is highlighted while locating the error.
https://www.loom.com/share/e163dcd48ef84992b9d4f2f13519815a?sid=0a841307-3548-434d-9f9a-c77532b06d44

**Comment by akash-codemonk** [comment]
> * [ ]  Header tab : Wrong key value pair is highlighted while locating the error.

Expected. We will go ahead with this. As this needs an alternative solution out of scope of the PR. The actual value exists in the first index so it navigates there. 

> * [ ]  Create any error on API end point to get execution error then click on error from error/logs tab, navigation is not working.
> * [ ]  Navigation to Body> JSON tab error not working
>   https://www.loom.com/share/31ad35aa0a584bbc994bb46a52303d35?sid=1cefdebd-6724-4eb6-a82d-403f1ff1b246

Expected. API end point errors will navigate to the api pane only not to any particular field as they don't have information about the field which has the error.

> * [ ]  Resolving the error is not clearing error log from debugger

Expected. Removal/addition of errors is out of scope of the PR. 

> * [ ]  Any changes in settings result in landing authentication tab
>   https://www.loom.com/share/c0d4bd1be087463f8592e08ac40da788?sid=a983c8c5-56d6-4f97-90b4-be37242cdcc3

Expected. This works when we are in view mode of the datasource or any other screen. When in edit mode of the datasource page, it takes over the navigation 

@kavitasmoolya

File: app/client/cypress/support/Pages/PropertyPane.ts [hunk 1]
```diff
@@ -527,6 +527,10 @@ export class PropertyPane {
     this.agHelper.AssertElementVisible(this._propertyControl(property));
   }
 
+  public AssertIfPropertyIsNotVisible(property: string) {
+    this.agHelper.AssertElementNotVisible(this._propertyControl(property));
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/PropertyPane.ts:531]: Use `agHelper.AssertElementAbsence`

File: app/client/cypress/e2e/Regression/ClientSide/Debugger/Api_pane_navigation_spec.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import * as _ from "../../../../support/Objects/ObjectsCore";
+import { EntityItems } from "../../../../support/Pages/AssertHelper";
+
+describe("excludeForAirgap", "Api pane navigation", () => {
+  it("Navigation to Grapql pagination field", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Debugger/Api_pane_navigation_spec.ts:5]: Index `it` blocks.

File: app/client/cypress/e2e/Regression/ClientSide/Debugger/Api_pane_navigation_spec.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import * as _ from "../../../../support/Objects/ObjectsCore";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Debugger/Api_pane_navigation_spec.ts:1]: Import only required objects from `ObjectsCore`

File: app/client/cypress/e2e/Regression/ClientSide/Debugger/Api_pane_navigation_spec.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import { EntityItems } from "../../../../support/Pages/AssertHelper";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Debugger/Api_pane_navigation_spec.ts:2]: Import `entityItems` also from `ObjectsCore`

File: app/client/cypress/support/Pages/PropertyPane.ts (hunk 1 above)
Comment (akash-codemonk) [reply app/client/cypress/support/Pages/PropertyPane.ts:531]: Hey no here the element exists, but I wanted to assert if the element is not visible. `AssertElementAbsence` checks if element exists in the dom.

Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/PropertyPane.ts:531]: Oh, you mean it exists will exist in DOM but not visible in UI?.. I see.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**